lib/ProductOpener/Display.pm

```diff
@@ -7739,31 +7739,25 @@ sub display_possible_improvement_description($$) {
 	my $tagid = shift;
 
 	my $html = "";
-
-	if ((defined $product_ref->{improvements_data}) and (defined $product_ref->{improvements_data}{$tagid})) {
-
-		# Comparison of product nutrition facts to other products of the same category
-
-		if ($tagid =~ /^en:nutrition-(very-)?high/) {
-			$html .= "<p>" . lang("value_for_the_product") . separator_before_colon($lc) . ": " . $product_ref->{improvements_data}{$tagid}{product_100g}
-			. "<br>" . sprintf(lang("value_for_the_category"), display_taxonomy_tag($lc, "categories", $product_ref->{improvements_data}{$tagid}{category}))
-			. separator_before_colon($lc) . ": " . $product_ref->{improvements_data}{$tagid}{category_100g}
-			. "</p>\n";
-		}
-
-		# Opportunities to improve the Nutri-Score by slightly changing the nutrients
-
-		if ($tagid =~ /^en:better-nutri-score/) {
-			# msgid "The Nutri-Score can be changed from %s to %s by changing the %s value from %s to %s (%s percent difference)."
-			$html .= "<p>" . sprintf(lang("better_nutriscore"),
-				uc($product_ref->{improvements_data}{$tagid}{current_nutriscore_grade}),
-				uc($product_ref->{improvements_data}{$tagid}{new_nutriscore_grade}),
-				lc(lang("nutriscore_points_for_" . $product_ref->{improvements_data}{$tagid}{nutrient})),
-				$product_ref->{improvements_data}{$tagid}{current_value},
-				$product_ref->{improvements_data}{$tagid}{new_value},
-				sprintf("%d", $product_ref->{improvements_data}{$tagid}{difference_percent})) . "</p>";
-		}
-	}
+	my $template_data_ref_improvement = {};
+
+	$template_data_ref_improvement->{tagid} = $tagid;
+	$template_data_ref_improvement->{product_ref_improvements_data} = $product_ref->{improvements_data};
+	$template_data_ref_improvement->{product_ref_improvements_data_tagid} = $product_ref->{improvements_data}{$tagid};
```

Comment:
If $product_ref->{improvements_data} is not defined, the line above will trigger an error.

That's why we had "	if ((defined $product_ref->{improvements_data}) and (defined $product_ref->{improvements_data}{$tagid})) {" in the Perl code before.